velox/type/Type.cpp

```diff
@@ -57,6 +57,7 @@ TypeKind mapNameToTypeKind(const std::string& name) {
       {"VARBINARY", TypeKind::VARBINARY},
       {"TIMESTAMP", TypeKind::TIMESTAMP},
       {"DATE", TypeKind::DATE},
+      {"DECIMAL", TypeKind::SHORT_DECIMAL},
```

Comment:
How will callers get `LONG_DECIMAL`? I feel we should use `DECIMAL`, and `DECIMAL128` for the names and enums as well. That way, we will have scope to add more variants (eg. `int32`) in the future. `SHORT_`, `LONG_` will limit our options to extend.

Reply:
I had a chat with Masha. The chances of adding another variant are low since that would involve adding a lot more specialization for the operators. Let's stick to current SHORT and LONG prefixes as they are more clear to read.

Reply:
But we agreed that this will have to be renamed to `SHORT_DECIMAL` and add another entry for `LONG_DECIMAL`.

Reply:
this mapping is used while resolving the string names to types. For decimal, if we want to support signatures of kind DECIMAL(x,y) then having DECIMAL and DECIMAL128 string names would be confusing.
My thought was while resolving the type, if DECIMAL is identified as SHORT_DECIMAL, we will allow the constructor to resolve the underlying storage type. If refer to the constructor, it infers it from precision. So doesn't matter whether DECIMAL string is mapped to SHORT or LONG in this mapping.